stellargraph/datasets/dataset_loader.py

```diff
@@ -0,0 +1,108 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2019-2020 Data61, CSIRO
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Enable easy loading of sample datasets for demonstrations
+"""
+
+import os
+from shutil import unpack_archive
+from urllib.request import urlretrieve
+from typing import List, Optional
+
+
+class DatasetLoader(object):
+    """ A class to download sample datasets.
+    the default path of ~/data can be changed by setting the STELLARGRAPH_DATASETS_PATH environment variable."""
+
+    def __init__(
+        self,
+        name: str,
+        directory_name: str,
+        url: str,
+        url_archive_format: Optional[str],
+        expected_files: List[str],
+        description: str,
+        source: str,
+        data_subdirectory_name: Optional[str] = None,
+    ) -> None:
+        self.name = name
+        self.directory_name = directory_name
+        self.url = url
+        self.url_archive_format = url_archive_format
+        self.expected_files = expected_files
```

Comment:
To save us from missing a file, should we validate:

```python
if url_archive_format is None and len(expected_files) != 1:
    raise ValueError("url_archive_format is None, which requires a single expected_file, found: {expected_files!r}")
```

Reply:
Great suggestion, and this now works well in __init_subclass__